### PR TITLE
docs: replace deprecated `Linter.FlatConfig` type with `Linter.Config`

### PR DESCRIPTION
**Pull request by caumeira**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

### CODE REVIEW

**Review by amareshsm — APPROVED** [review]
LGTM,

Awaiting a review from a member of the ESLint Team.

**Review by nzakas — APPROVED** [review]
LGTM. Holding off on merging while we determine the patch release status.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
